wallet-new/src/Cardano/Wallet/API/V1/Migration/Types.hs

```diff
@@ -195,7 +195,7 @@ instance Migrate V0.AccountId (V1.WalletId, V1.AccountIndex) where
         (,)
             <$> eitherMigrate (V0.aiWId accId)
             <*> first
-                    Errors.MigrationFailed
+                    (Errors.MigrationFailed . sformat build)
```

Comment:
do you think it makes sense to introduce new constructor in `WalletError` to capture exact underying issue (instead of wrapping it under `MigrationFailed`). I wonder does it makes sense to have this pattern in general.

This would make it possible to have more detailed error handling if it is needed. Now we mask underlying errors under higher level error (which is `MigrationFailed` in this case).

I wonder do we have some opinion about this?